jctools-core/src/test/java/org/jctools/queues/atomic/ScLinkedAtomicQueueTest.java

```diff
@@ -0,0 +1,75 @@
+package org.jctools.queues.atomic;
+
+import org.junit.Test;
+
+import java.util.Queue;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public abstract class ScLinkedAtomicQueueTest {
+    protected abstract Queue<Integer> newQueue();
+
+    @Test
+    public void removeWorksInBeginMiddleEnd() throws InterruptedException {
```

Comment:
That's fine, but I would prefer to have some simple tests on top of concurrent tests, so we can debug simple bugs simply.